lib/config/error-codes/index.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const errors = require('kuzzle-common-objects').errors;
+const api = require('./api');
+const assert = require('assert');
+
+/** Check the format of the error-codes config files
+ *  @param {object} - error config file domain
+ */
+
+function checkErrorCodes(errorCodes) {
+
+  for (const domain of Object.keys(errorCodes)) {
+    const domainName = errorCodes[domain];
+    
+    assert(domainName.hasOwnProperty('subdomains'), `Error configuration file : Missing required 'subdomains' field in '${domain}'.`);
+    assert(typeof domainName.subdomains === 'object', `Error configuration file : Field 'subdomains' must be an object in '${domain}'.`);
```

Comment:
I didn't understand why `typeof` is not accurate here

Reply:
Because you want to know if your variable is an plain object `{ }`, and in JS everything is an "object"
![image](https://user-images.githubusercontent.com/4447392/58970563-fc1d7800-87b9-11e9-821b-2aafbaef04ac.png)

Reply:
the following returns `'object'` (this list is not exhaustive):

```
typeof null
typeof []
typeof new Set()
typeof new Map()
typeof new WeakMap()
typeof new WeakSet()
typeof Buffer.from('foobar')
typeof new Uint8Array([1, 2, 3])
```